src/main/scala/com/github/simplesteph/ksm/parser/YamlAclParser.scala

```diff
@@ -0,0 +1,156 @@
+package com.github.simplesteph.ksm.parser
+
+import java.io.Reader
+
+import cats.data.Validated
+import cats.data.Validated.Valid
+import cats.implicits._
+import com.github.simplesteph.ksm.source.SourceAclResult
+import io.circe.generic.auto._
+import io.circe.syntax._
+import io.circe.yaml
+import kafka.security.auth._
+import org.apache.kafka.common.resource.PatternType
+import org.apache.kafka.common.security.auth.KafkaPrincipal
+
+import scala.collection.immutable._
+import scala.util.Try
+
+/**
+  * Parser that assumes that all ACLs are flattened
+  * and live under a YML format.
+  */
+class YamlAclParser() extends AclParser {
+
+  val name: String = "Yaml"
+
+  case class AclResources(groups: Option[Map[String, List[String]]],
+                          topics: Option[Map[String, List[String]]],
+                          transactional_ids: Option[Map[String, List[String]]],
+                          clusters: Option[Map[String, List[String]]])
+
+  case class AclYaml(users: Map[String, AclResources])
+
+  def getResource(resourceName: String, resourceType: ResourceType): Resource = {
+
+    if (resourceName.trim.endsWith("*") && resourceName.trim != "*")
+      new Resource(resourceType, resourceName.trim.dropRight(1), PatternType.PREFIXED)
+    else new Resource(resourceType, resourceName.trim, PatternType.LITERAL)
+  }
+
+  def parseResource(principal: KafkaPrincipal,
+                    resourceType: ResourceType,
+                    resources: Map[String, List[String]]
+                   ): SourceAclResult = {
+
+    val parsed = resources.toList.flatMap({
+      case (resourceName, resourceAcls) => {
+        val resource = getResource(resourceName, resourceType)
+        resourceAcls.map(acl => parseAcl(principal, resource, acl))
+      }
+    })
+
+    val acls = parsed.filter(_.isValid).flatMap(_.getOrElse(List())).toSet
+    val errs = parsed.filter(_.isInvalid).map(_.toEither.left.get).map(err => Try(err))
+    SourceAclResult(acls, errs)
+  }
+
+  val clusterWildcard = Resource(Cluster, "*", PatternType.LITERAL)
+
+  def parseAcl(principal: KafkaPrincipal, resource: Resource, acl: String): Validated[YamlParserException, List[(Resource, Acl)]] = {
+    acl.split(",") match {
+      case Array(operation, permissionType, host) =>
+        Try {
+          List(
+            (resource, Acl(principal, PermissionType.fromString(permissionType), host, Operation.fromString(operation)))
+          )
+        }
+          .toEither
+          .leftMap(error => new YamlParserException(s"Could not parse ACL '$acl' for principal '$principal' and resource '$resource'", error))
+          .toValidated
+      case Array(abbrev) => (resource.resourceType, abbrev.toLowerCase()) match {
+        case (Topic, "admin") =>
+          Valid(
+            List(
+              (resource, Acl(principal, Allow, "*", Describe)),
```

Comment:
Isn't there an `All` for that reason? 
cf https://github.com/simplesteph/kafka-security-manager/blob/master/example/acls.csv

Reply:
Will remove